### PR TITLE
Fixes for perl module package

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR does some fixing up of the perl-atlas-modules Conda package:

 - Firstly I've added perl-params-validate Conda dependency (was failing to build from CPAN but works this way). This was all that was needed to get the package to build.

 - Secondly, I've added a bunch more dependencies to the build. The way this package has been set up to work, there are a couple of dependencies that have to come directly from CPAN, and have to be bundled into the package. I've looked at making the recipes for those and there does seem to be some dependency issues that makes it a non-trivial task so we'll have to keep packaging those for now. However a bunch of upstream dependencies for these packages ARE in Bioconda, and by adding those as conda dependencies we minimise the amount we're adding to the package. 

 - I've also bumped the version. The current 0.1.3 was clearly out of date (I can see a 0.1.4 in Anaconda, though only for OSX https://anaconda.org/ebi-gene-expression-group/perl-atlas-modules/files), and I've bumped the build to reflect the changes I've made here.

 - I've flipped the maintainer to myself

### CODE REVIEW

**Comment by pinin4fjords** [comment]
Oh dammit, hadn't seen https://github.com/ebi-gene-expression-group/atlas-conda/pull/17. Revising now...

**Comment by pcm32** [comment]
Great work @pinin4fjords , is this working on the mac as well? We can live without that if we start using containers though after moving it to bioconda (although they might not merge it until the macOS goes through).

**Comment by pinin4fjords** [comment]
Now incorporated the minor dependency additions from https://github.com/ebi-gene-expression-group/atlas-conda/pull/17, didn't see any harm in doing so if @suhaibMo thought they would be required. And yes @pcm32 , this does build on macOS too.

File: recipes/perl-atlas-modules/build.sh [hunk 1]
```diff
@@ -16,16 +17,19 @@ PERLLIB="${PREFIX}/lib/perl5/${perl_version}/perl_lib"
...
+                    Bio::MAGETAB \
+                    Date::Parse \
+                    Test::MockObject \
+                    Text::TabularDisplay \
+                    Log::Dispatch::File \
+                    IO::CaptureOutput \
+                    Class::DBI
 
 mkdir -p ${PREFIX}/etc/conda/activate.d/
 echo "export export PERL5LIB=$PERL5LIB:$atlasprodDir/perl_modules:$PERLLIB/lib/perl5" > ${PREFIX}/etc/conda/activate.d/${PKG_NAME}-${PKG_VERSION}.sh
```
Comment (pcm32) [inline recipes/perl-atlas-modules/build.sh]:
```suggestion
echo "export PERL5LIB=$PERL5LIB:$atlasprodDir/perl_modules:$PERLLIB/lib/perl5" > ${PREFIX}/etc/conda/activate.d/${PKG_NAME}-${PKG_VERSION}.sh
```
I wonder if the extra export was avoiding this from being exported or not having any impact.

**Comment by pcm32** [comment]
Great that you could include so many packages to the meta.yaml instead of in the build! Might be good to try a few of the curators scripts to make sure it works before releasing/pushing, and then we could add to bioconda to gain the containers.

File: recipes/perl-atlas-modules/build.sh (hunk 1 above)
Comment (pinin4fjords) [reply recipes/perl-atlas-modules/build.sh]: Thanks Pablo, saw this earlier but forgot to correct it

**Comment by pinin4fjords** [comment]
For completeness I'm setting this to the same as the bioconda port (see https://github.com/bioconda/bioconda-recipes/pull/26559).

I'm not going to make new builds for this on our channel, to avoid any conflicts with the Bioconda one.

**Pull request merged**